guides/type_definitions/arguments.md

```diff
@@ -0,0 +1,130 @@
+---
+layout: guide
+doc_stub: false
+search: true
+section: Type Definitions
+title: Arguments
+desc: Fields may take arguments as inputs
+index: 10
+class_based_api: true
+---
+
+Fields can take **arguments** as input. These can be used to determine the return value (eg, filtering search results) or to modify the application state (eg, updating the database in `MutationType`).
+
+Arguments are defined with the `argument` helper. These arguments are passed as [keyword arguments](https://robots.thoughtbot.com/ruby-2-keyword-arguments) to the resolver method:
+
+```ruby
+field :search_posts, [PostType], null: false do
+  argument :category, String, required: true
+end
+
+def search_posts(category:)
+  Post.where(category: category).limit(10)
+end
+```
+
+To make an argument optional, set `required: false`, and set default values for the corresponding keyword arguments:
+
+```ruby
+field :search_posts, [PostType], null: false do
+  argument :category, String, required: false
+end
+
+def search_posts(category: nil)
+  if category
+    Post.where(category: category).limit(10)
+  else
+    Post.all.limit(10)
+  end
+end
+```
+
+Be aware that if all arguments are optional and the query does not provide any arguments, then the resolver method will be called with no arguments. To prevent an `ArgumentError` in this case, you must either specify default values for all keyword arguments (as done in the prior example) or use the double splat operator argument in the method definition. For example:
+
+```ruby
+def search_posts(**args)
+  if args[:category]
+    Post.where(category: args[:category]).limit(10)
+  else
+    Post.all.limit(10)
+  end
+end
+```
+
+Another approach is to use `default_value: value` to provide a default value for the argument if it is not supplied in the query.
+
+```ruby
+field :search_posts, [PostType], null: false do
+  argument :category, String, required: false, default_value: "Programming"
+end
+
+def search_posts(category:)
+  Post.where(category: category).limit(10)
+end
+```
+
+Use `as: :alternateName` to use a different key from within your resolvers while
```

Comment:
I don't recommend using camel-case at all in Ruby code for GraphQL-Ruby, can we change this example to `as: :alternate_name`?